account_invoice_change_currency/models/account_change_currency.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2017-2018 Vauxoo
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+    original_price_unit = fields.Monetary(
+        default=lambda self: self.price_unit,
+        help="""
+        Save price unit from every line in the invoice when the currency back to
+        original state""",
+    )
+
+    def create(self, values):
+        lines = super().create(values)
+        for line in lines:
+            line.write({"original_price_unit": line.price_unit})
+        return lines
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    custom_rate = fields.Float(
+        digits=(12, 6),
+        default=1,
+        help="Set new currency rate to apply on the invoice\n."
+        "This rate will be taken in order to convert amounts between the "
+        "currency on the invoice and last currency",
+    )
+    created_invoice_currency = fields.Many2one(
+        "res.currency",
+        help="""Save the original currency from the company""",
```

Comment:
Company has nothing to do with the currency, it could be any currency

Reply:
Done!